hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlClientTest.java

```diff
@@ -108,4 +117,42 @@ public void when_resultClosed_then_jobCancelled_withNoResults() {
         logger.info("after res.close() returned");
         assertJobStatusEventually(job, FAILED);
     }
+
+    // test for https://github.com/hazelcast/hazelcast/issues/19897
+    @Test
+    public void when_resultClosed_then_executionContextCleanedUp() {
+        HazelcastInstance client = factory().newHazelcastClient();
+        SqlService sql = client.getSql();
+
+        IMap<Integer, Integer> map = instance().getMap("map");
+        Map<Integer, Integer> tmpMap = new HashMap<>();
+        for (int i = 0; i < 100_000; i++) {
+            tmpMap.put(i, i);
+            if (i % 10_000 == 0) {
+                map.putAll(tmpMap);
+                tmpMap.clear();
+            }
+        }
+
+        createMapping("map", Integer.class, Integer.class);
+
+        for (int i = 0; i < 100; i++) {
+            SqlResult result = sql.execute("SELECT * FROM map");
+            result.close();
+        }
+
+        JetServiceBackend jetService = getJetServiceBackend(instance());
+        Collection<ExecutionContext> contexts = jetService.getJobExecutionService().getExecutionContexts();
+        // assert that all ExecutionContexts are eventually cleaned up
```

Comment:
minor: could you also note that this assertion passes only if handling time of a network package from old execution doesn't exceed `JobExecutionService#FAILED_EXECUTION_EXPIRY_NS`(although this is not possible under normal load conditions)?